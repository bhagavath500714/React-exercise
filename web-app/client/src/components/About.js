import React from 'react'

export default function About() {
    const foTest = [
        {id: 1, title: 'title-1', completed: false },
        {id: 2, title: 'title-2', completed: true }
      ];
  
      // {forTest.map((todo) => {
      //   return (<Todo todo={todo}/>)
      // })}
      const RecordData = ({todo}) => {
          const { id, title, completed } = todo;
          const h1 = <h1>{title}</h1>
          const text = completed ? <strike>{h1}</strike> : h1;
          return <div data-testid={`todo-${id}`}>{ text }</div>
    }
    return (
        <div>
            {foTest.map((item) => {
                return (<RecordData todo={item} key={item.id}/>)
            })}
        </div>
    )

    
}
