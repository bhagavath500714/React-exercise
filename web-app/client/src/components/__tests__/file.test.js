import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'
import About from '../About';

afterEach(() => {
    cleanup();
});

test('should render non-completed todo', () => {
    const todo = { id: 1, title: 'wash dishes', completed: false };
    render(<About todo={todo} />);
    const todoElement = screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('title-1');
});

test('should render completed todo', () => {
    const todo = { id: 2, title: 'wash car', completed: true };
    render(<About todo={todo} />);
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('title-2');
});

test('matches snapshot', () => {
    const todo = { id: 2, title: 'wash car', completed: true };
    const tree = renderer.create(<About todo={todo}/>).toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
});



// expect(todoElement).not.toContainHTML('<strike>');

// test('test', () => {
//     expect(true).toBe(true);
// })

// npm i --save-dev @testing-library/react react-test-render
// https://www.youtube.com/watch?v=ML5egqL3YFE