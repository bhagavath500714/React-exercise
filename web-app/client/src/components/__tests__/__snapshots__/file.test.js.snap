// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    data-testid="todo-1"
  >
    <h1>
      title-1
    </h1>
  </div>
  <div
    data-testid="todo-2"
  >
    <strike>
      <h1>
        title-2
      </h1>
    </strike>
  </div>
</div>
`;
